spec/formatting.md

```diff
@@ -214,9 +214,21 @@ the following steps are taken:
 
 3. Perform _option resolution_.
 
-4. Call the function implementation with the following arguments:
+4. Determine the **_<dfn>function context</dfn>_** for calling the function implementation.
+   This includes:
 
-   - The current _locale_.
+   - The current _locale_,
+     potentially including a fallback chain of locales.
+   - The base directionality of the _message_ and its _text_ tokens.
+
+   If the resolved mapping of _options_ includes any `u:` options
```

Comment:
This seems like special pleading on our part. The `u:` namespace is "just a namespace"?

Perhaps:

```suggestion
Implementations are encouraged to support _options_ defined in
the Unicode Reserved Namespace (`u:`).
```

Reply:
It's not "just a namespace", because it needs special powers to affect the _function context_, though.

Reply:
I think that the "function context" context distinction is an implementation detail.

In most formatters implementations the locale and the options on how to format are passed to the constructor as parameters, at the same time.

I would rather look at this as "universal function parameters" that might be recognized and honored by several / all functions.

Reply:
I think the problem here is that the **_current set_** of `u:` options behaves that way, but we might introduce one that isn't function context affecting in the future? If our intention to is to _require_ that all such options be context-affecting, we should set that as a requirement in the design doc and elsewhere. We might need to contemplate an additional namespace in the future as well, although I can't think of any universal options just at the moment that we wouldn't just put in the default namespace.

Note: I am not disagreeing with doing this. Just making sure we're consistent and clear about it.